dokuwiki-text/src/main/java/org/xwiki/contrib/dokuwiki/text/internal/input/DokuWikiInputFilterStream.java

```diff
@@ -282,29 +282,28 @@ private void readDocument(File directory, String dokuwikiDataDirectory, DokuWiki
 
-                    //Wiki document revision
-
-                    if ((documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).containsKey(KEY_CREATED)
-                            && documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).containsKey(KEY_MODIFIED))
-                            && documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).getLong(KEY_CREATED)
-                            < documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).getLong(KEY_MODIFIED))
-                    {
-                        //Wiki Document Locale
-                        proxyFilter.beginWikiDocumentLocale(Locale.ROOT, documentLocaleParameters);
-                        //read revisions
-                        readPageRevision(file, dokuwikiDataDirectory, proxyFilter);
-                    } else {
-
-                        String pageContents = FileUtils.readFileToString(file, StandardCharsets.UTF_8);
-                        String convertedContent = parseContent(pageContents);
-                        documentLocaleParameters.put(WikiDocumentFilter.PARAMETER_CONTENT, convertedContent);
-                        //Wiki Document Locale
-                        proxyFilter.beginWikiDocumentLocale(Locale.ROOT, documentLocaleParameters);
-                        readAttachments(pageContents, file, dokuwikiDataDirectory, proxyFilter);
+                    try{
+                        String metadataFileContents = FileUtils.readFileToString(new File(fileMetadataPath), "UTF-8");
+                        MixedArray documentMetadata = Pherialize.unserialize(metadataFileContents).toArray();
+                        readDocumentParametersFromMetadata(documentMetadata, documentLocaleParameters);
+
+                        //Wiki document revision
+
+                        if ((documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).containsKey(KEY_CREATED)
+                                && documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).containsKey(KEY_MODIFIED))
+                                && documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).getLong(KEY_CREATED)
+                                < documentMetadata.getArray(KEY_CURRENT).getArray(KEY_DATE).getLong(KEY_MODIFIED))
+                        {
+                            //Wiki Document Locale
+                            proxyFilter.beginWikiDocumentLocale(Locale.ROOT, documentLocaleParameters);
+                            //read revisions
+                            readPageRevision(file, dokuwikiDataDirectory, proxyFilter);
+                        } else {
+                            documentLocaleParameters = readDocument(file,documentLocaleParameters, dokuwikiDataDirectory, proxyFilter);
+                        }
+                    } catch (Exception e ){
+                        documentLocaleParameters = readDocument(file,documentLocaleParameters, dokuwikiDataDirectory, proxyFilter);
```

Comment:
Since the couldn't read the meta file for some reason, it makes sense to move on to other files, but you should probably log a warning and the error.